src/cpp/src/visual_language/clip.cpp

```diff
@@ -76,7 +76,12 @@ void bicubic_resize(const clip_image_u8 &img, clip_image_u8 &dst, int target_wid
 
     dst.nx = target_width;
     dst.ny = target_height;
-    dst.buf.resize(3 * target_width * target_height);
+    const int target_size = 3 * target_width * target_height;
+    dst.buf.resize(target_size);
+    if (img.nx == target_width && img.ny == target_height) {
+        std::memcpy(dst.buf.data(), img.buf.data(), target_size);
```

Comment:
not good here. `target_size` is a number of elements. Please multiply it by a bytesize of one element